### PR TITLE
build: fix build error on Visual Studio 2013 and earlier

### LINKED ISSUE #11979: Stop support VS2013?
Start using `noexcept` in commit 43f821afb91330958d4723f6300a5ec0bfa2185e makes it impossible to use VS2013.

### LINKED ISSUE #11999: OpenCV master build fails on Visual Studio 2013 and earlier
##### System information (version)

- OpenCV => recent master (technically 43f821afb / #11899  and after )
- Operating System / Platform => Windows 10 64bit
- Compiler => Visual Studio 2013  64bit (and probably earlier)

##### Detailed description
We get dozens of error about ```noexcept``` when building with Visual Studio 2013 and earlier.
```
  D:\Jenkins\workspace\OpenCV-MSBuild\modules\core\include\opencv2/core/types.hpp(166): error C3646: 'noexcept' : unknown override specifier (D:\Jenkins\workspace\OpenCV-MSBuild\modules\imgproc\src\resize.sse4_1.cpp) [D:\Jenkins\workspace\OpenCV-MSBuild\build\modules\imgproc\opencv_imgproc_SSE4_1.vcxproj]
```

I googled a bit and figured out that ```noexcept``` is supported only from Visual Studio 2015 and later. 
[answer on StackOverflow](https://stackoverflow.com/questions/18387640/how-to-deal-with-noexcept-in-visual-studio/33298731#33298731)
[commnet on StackOverflow](https://stackoverflow.com/questions/18387640/how-to-deal-with-noexcept-in-visual-studio#comment27004724_18387764) 
[MSDN](https://msdn.microsoft.com/en-us/library/wfa0edys.aspx)

I'll throw a workaround patch later.

##### Steps to reproduce

Build recent master using Visual Studio 2013 or earlier

Update: my mistake.  Only happens on master, not on 3.4

### PR DESCRIPTION
**Pull request by tomoaki0705**
resolves #11999
resolves #11979

### CODE REVIEW

**Comment by tomoaki0705** [comment]
Expected question
1. branch
  This PR is specific for master.  I once tried to merge with 3.4, but it was just causing a mess.  The original PR #11899 was also merging to master, so this PR should follow the same way.
2. Where to workaround
  This is kind of easiest workaround. We could do a smarter way like the ```inline``` of the compiler.  We could add a test script in CMake to check whether the compiler accepts ```noexcept``` identifier or not.  Open for discussion.

File: modules/core/include/opencv2/core/types.hpp [hunk 1]
```diff
@@ -57,6 +57,12 @@
 #include "opencv2/core/cvstd.hpp"
 #include "opencv2/core/matx.hpp"
 
+#if defined(_MSC_VER) && (_MSC_VER <= 1800)
+#define CV_NOEXCEPT
+#else
+#define CV_NOEXCEPT noexcept
+#endif
```
Comment (alalek) [inline modules/core/include/opencv2/core/types.hpp]: Could you move this into cvdef.h file please?
Similar to [this](https://github.com/opencv/opencv/blob/3.4.2/modules/core/include/opencv2/core/cvdef.h#L440-L453)

**Comment by alalek** [comment]
@tomoaki0705 I pushed an update to this patch. Please take a look.

**Comment by tomoaki0705** [comment]
Thanks @alalek. it looks much better

**Pull request merged**